### PR TITLE
Allow passing custom parameters when creating Zip FS

### PR DESCRIPTION
**Pull request by geoand**
- Relates to: https://github.com/quarkusio/quarkus/issues/38128

### CODE REVIEW

**Comment by geoand** [comment]
cc @aloubyansky

File: src/main/java/io/quarkus/fs/util/ZipUtils.java [hunk 1]
```diff
@@ -114,20 +114,31 @@ public static void zip(Path src, Path zipFile) throws IOException {
...
+            effectiveEnv = new HashMap<>(CREATE_ENV);
             // explicitly create any parent dirs, since the ZipFileSystem only creates a new file
             // with "create" = "true", but doesn't create any parent dirs.
 
             // It's OK to not check the existence of the parent dir(s) first, since the API,
             // as per its contract doesn't throw any exception if the parent dir(s) already exist
             Files.createDirectories(zipFile.getParent());
         }
+        if (env != null) {
+            effectiveEnv.putAll(env);
```
Comment (aloubyansky) [inline src/main/java/io/quarkus/fs/util/ZipUtils.java:138]: How about creating a copy only when env isn't null and isn't empty?

Comment (geoand) [reply src/main/java/io/quarkus/fs/util/ZipUtils.java:138]: Makes sense

**Review by aloubyansky — APPROVED** [review]

**Comment by geoand** [comment]
Thanks! 

Any chance I could get a release tomorrow?

**Comment by aloubyansky** [comment]
Hopefully, it's released. I had some timeouts from Sonatype and weird log in related errors, let's see whether it ends up on Central.

**Comment by geoand** [comment]
Thanks a lot!

I'll check tomorrow morning

**Comment by aloubyansky** [comment]
Looks like it made it https://repo1.maven.org/maven2/io/quarkus/quarkus-fs-util/0.0.10/

**Comment by geoand** [comment]
👌

**Pull request merged**